cdap-common/src/main/java/co/cask/cdap/common/kerberos/SecurityUtil.java

```diff
@@ -167,4 +169,29 @@ public void run() {
         }, delaySec, delaySec, TimeUnit.SECONDS);
     }
   }
+
+  /**
+   * Returns a {@link KerberosName} from the given {@link KerberosPrincipalId} if the given kerberos principal id
+   * is valid. Refer to
+   * <a href="https://web.mit.edu/kerberos/krb5-1.5/krb5-1.5.4/doc/krb5-user/What-is-a-Kerberos-Principal_003f.html">
+   * Kerberos Principal</a> for details.
+   *
+   * @param principalId The {@link KerberosPrincipalId} from which {@link KerberosName} needs to be created
+   * @return {@link KerberosName} for the given {@link KerberosPrincipalId}
+   */
+  public static KerberosName getKerberosName(KerberosPrincipalId principalId) {
+    return new KerberosName(principalId.getPrincipal());
+  }
+
+
+  /**
+   * Checks if the given {@link KerberosPrincipalId} is valid or not by calling
+   * {@link #getKerberosName(KerberosPrincipalId)}. This is just a wrapper around
+   * {@link #getKerberosName(KerberosPrincipalId)} to not return an object to the caller for simplicity.
+   *
+   * @param principalId {@link KerberosPrincipalId} which needs to be validated
+   */
+  public static void validateKerberosPrincipal(KerberosPrincipalId principalId) {
```

Comment:
Added the doc for throws. 

Added this method just as more of helper to not return the object to the called if only validation is needed.